src/core/time_arrays.jl

```diff
@@ -56,11 +56,17 @@ function _single_index_check(data, idxs)
 	end
 end
 
-# Helper to print stacktrace
+# Helper to print stacktrace for the integer indexing deprecatino warning
 function _get_stacktrace_string()
 	s = ""
 	for line in stacktrace()
-		s = string(s, line, "\n")
+		line_string = string(line)
+		if startswith(line_string, "run_timestep") # shorten the reporting on calls to run_timestep
+			l = "run_timestep function call"
```

Comment:
My preference would be to just `return s` here. If it hits a "run_timestep" line, then the previous line would be the "macro expansion" line which has the actual component line number info of where the error is happening

Reply:
@ckingdon95 sounds good to me, done!